src/psyclone/gocean1p0.py

```diff
@@ -460,6 +465,101 @@ def __init__(self, parent=None,
         if not GOLoop._bounds_lookup:
             GOLoop.setup_bounds()
 
+    # -------------------------------------------------------------------------
+    def _halo_read_access(self, arg):
+        '''Determines whether the supplied argument has (or might have) its
+        halo data read within this loop. Returns True if it does, or if
+        it might and False if it definitely does not.
+
+        :param arg: an argument contained within this loop
+        :type arg: :py:class:`psyclone.dynamo0p3.DynArgument`
+        :return: True if the argument reads, or might read from the \
+                 halo and False otherwise.
+        :rtype: bool
+
+        '''
+        if arg.type != 'field':
```

Comment:
You could do `return arg.type == 'field'` here.

Reply:
Also, how come you don't check whether the argument is read or written? (Apologies if I'm missing something...)

Reply:
You are right about both (previously I deleted the only-written HaloEx fields afterwards, but since this method is called _halo_read_access it makes more sense to filer them here)